presto-main/src/main/java/com/facebook/presto/operator/scalar/DateTimeFunctions.java

```diff
@@ -180,9 +181,15 @@ public static double toUnixTimeFromTimestampWithTimeZone(@SqlType(StandardTypes.
     // the maximum year represented by 64bits timestamp is ~584944387 it may require up to 35 characters.
     public static Slice toISO8601FromTimestamp(ConnectorSession session, @SqlType(StandardTypes.TIMESTAMP) long timestamp)
     {
-        DateTimeFormatter formatter = ISODateTimeFormat.dateTime()
-                .withChronology(getChronology(session.getTimeZoneKey()));
-        return utf8Slice(formatter.print(timestamp));
+        if (session.isLegacyTimestamp()) {
+            DateTimeFormatter formatter = ISODateTimeFormat.dateTime()
+                    .withChronology(getChronology(session.getTimeZoneKey()));
+            return utf8Slice(formatter.print(timestamp));
+        }
+        else {
+            // FIXME: Simply remove this function when legacyTimestamp flag will be dropped
+            throw new PrestoException(FUNCTION_NOT_FOUND, "to_iso8601(timestamp) does not exist");
```

Comment:
I have serious doubts about wikipedia reliablity based on other sources including Java libraries implementations. I've requested access to ISO 8601 to confirm that, but for now leaving in "as is" state.